tests/test_helpers.py

```diff
@@ -29,14 +29,26 @@ def test_initials_returns_first_and_middle():
 
 
 @freeze_time("2023-08-18")
-def test_sns_log(caplog, stubbed_sns_client):
+def test_sns_log_publishes_status_messages_fail(stubbed_sns_client_start_fail):
     config_values = load_config_values()
     with patch("boto3.client") as mocked_boto_client:
-        mocked_boto_client.return_value = stubbed_sns_client
-        sns_log(config_values, status="start")
+        mocked_boto_client.return_value = stubbed_sns_client_start_fail
 
-        sns_log(config_values, status="success")
-        assert "Carbon run has successfully completed." in caplog.text
+        sns_start_response = sns_log(config_values, status="start")
+        assert sns_start_response["MessageId"] == "StartMessageId"
 
-        sns_log(config_values, status="fail")
-        assert "Carbon run has failed." in caplog.text
+        sns_fail_response = sns_log(config_values, status="fail")
+        assert sns_fail_response["MessageId"] == "FailMessageId"
+
+
+@freeze_time("2023-08-18")
+def test_sns_log_publishes_status_messages_success(stubbed_sns_client_start_success):
+    config_values = load_config_values()
+    with patch("boto3.client") as mocked_boto_client:
+        mocked_boto_client.return_value = stubbed_sns_client_start_success
+
+        sns_start_response = sns_log(config_values, status="start")
+        assert sns_start_response["MessageId"] == "StartMessageId"
+
+        sns_fail_response = sns_log(config_values, status="success")
+        assert sns_fail_response["MessageId"] == "SuccessMessageId"
```

Comment:
I think it's nice these are broken into [start, succes] and [start, fail].  But wonder if it's worth going a step further and just have each tested?  Something along the lines of

  * `stubbed_sns_client_start` (fixture) --> `test_sns_log_publishes_status_message_start`
  * `stubbed_sns_client_success` (fixture) --> `test_sns_log_publishes_status_message_success`
  * `stubbed_sns_client_fail` (fixture) --> `test_sns_log_publishes_status_message_fail`

In that case, I think it's still worth keeping the `stubbed_sns_client_start_success` and `stubbed_sns_client_start_fail` fixtures, as they are helpful for testing the actual app flow and logic.

Reply:
@ghukill Thank you for all your help and suggestions to improve testing! Please see the latest commit for updates to tests related to SNS logging and failed runs.